core/matrix/csr.cpp

```diff
@@ -534,6 +539,28 @@ bool Csr<ValueType, IndexType>::is_sorted_by_column_index() const
 }
 
 
+template <typename ValueType, typename IndexType>
+std::unique_ptr<Csr<ValueType, IndexType>>
+Csr<ValueType, IndexType>::create_submatrix(const gko::span& row_span,
+                                            const gko::span& column_span) const
+{
+    using Mat = Csr<ValueType, IndexType>;
+    auto exec = this->get_executor();
+    auto sub_mat_size = gko::dim<2>(row_span.length(), column_span.length());
+    Array<size_type> row_nnz(exec, row_span.length());
+    exec->run(csr::make_fill_array(row_nnz.get_data(), row_nnz.get_num_elems(),
+                                   zero<size_type>()));
+    exec->run(csr::make_calculate_nonzeros_per_row_in_span(
+        this, row_span, column_span, &row_nnz));
+    auto sub_mat_nnz = reduce(row_nnz);
+    auto sub_mat =
+        Mat::create(exec, sub_mat_size, sub_mat_nnz, this->get_strategy());
+    exec->run(csr::make_compute_submatrix(this, &row_nnz, row_span, column_span,
+                                          sub_mat.get()));
+    return sub_mat;
```

Comment:
are we building the strategy data somewhere?
```suggestion
    sub_mat->make_srow();
    return sub_mat;
```